src/isomorphic/ReactDebugTool.js

```diff
@@ -49,6 +101,95 @@ var ReactDebugTool = {
       }
     }
   },
+  beginProfiling() {
+    if (__DEV__) {
+      if (isProfiling) {
+        return;
+      }
+
+      isProfiling = true;
+      flushHistory.length = 0;
+      resetMeasurements();
+    }
+  },
+  endProfiling() {
+    if (__DEV__) {
+      if (!isProfiling) {
+        return;
+      }
+
+      isProfiling = false;
+      resetMeasurements();
+    }
+  },
+  getFlushHistory() {
+    if (__DEV__) {
+      return flushHistory;
+    }
+  },
+  onBeginFlush() {
```

Comment:
I think this comes down to the way `printWasted()` heuristic works. It says “count a render as wasted if no DOM operations were made during the same batch”. I was trying to replicate the existing behavior so this is the main reason.

I think we may want to reconsider this as part of #6632. I’d leave it as is for now though.